crates/nu-command/src/platform/input/input_.rs

```diff
@@ -59,114 +68,96 @@ impl Command for Input {
-        let bytes_until: Option<String> = call.get_flag(engine_state, stack, "bytes-until-any")?;
-        let suppress_output = call.has_flag(engine_state, stack, "suppress-output")?;
-        let numchar: Option<Spanned<i64>> = call.get_flag(engine_state, stack, "numchar")?;
-        let numchar: Spanned<i64> = numchar.unwrap_or(Spanned {
-            item: i64::MAX,
-            span: call.head,
-        });
-
-        let from_io_error = IoError::factory(call.head, None);
-
-        if numchar.item < 1 {
-            return Err(ShellError::UnsupportedInput {
-                msg: "Number of characters to read has to be positive".to_string(),
-                input: "value originated from here".to_string(),
-                msg_span: call.head,
-                input_span: numchar.span,
-            });
+        // Those options are not supported by reedline, default to the legacy
+        // implementation
+        let use_legacy = [
+            call.get_flag::<String>(engine_state, stack, "bytes-until-any")?
+                .is_some(),
+            call.has_flag(engine_state, stack, "suppress-output")?,
+            call.get_flag::<Spanned<i64>>(engine_state, stack, "numchar")?
+                .is_some(),
+        ]
+        .iter()
+        .any(|x| *x);
+
+        if use_legacy {
+            return self.legacy_input(engine_state, stack, call, input);
         }
 
+        let prompt_str: Option<String> = call.opt(engine_state, stack, 0)?;
         let default_val: Option<String> = call.get_flag(engine_state, stack, "default")?;
-        if let Some(prompt) = &prompt {
-            match &default_val {
-                None => print!("{prompt}"),
-                Some(val) => print!("{prompt} (default: {val})"),
+        let history_file_val: Option<String> =
+            call.get_flag(engine_state, stack, "history-file")?;
+        let max_history: usize = call
+            .get_flag::<i64>(engine_state, stack, "max-history")?
+            .map(|l| if l < 0 { 0 } else { l as usize })
+            .unwrap_or(HISTORY_SIZE);
+
+        let default_str = match (&prompt_str, &default_val) {
+            (Some(_prompt), Some(val)) => format!("(default: {val}) "),
+            _ => "".to_string(),
+        };
+
+        let history_entries = match input {
+            PipelineData::Value(Value::List { vals, .. }, ..) => Some(vals),
+            _ => None,
+        };
+
+        // If we either have history entries or history file, we create an history
+        let history = match (history_entries.is_some(), history_file_val.is_some()) {
+            (false, false) => None, // Neither are set, no need for history support
+            _ => {
+                let mut history = match history_file_val {
+                    Some(file) => FileBackedHistory::with_file(max_history, file.into()),
+                    None => FileBackedHistory::new(max_history),
+                }
+                .expect("Error creating history file");
```

Comment:
I think this can't be `expect`ed and needs user facing error handling with `ShellError`. Commands are not permitted to panic.

Reply:
sounds good! Will fix!

Reply:
Thanks!